### PR TITLE
Add more proc-macro attribute tests

### PR DESCRIPTION
**Pull request by Aaron1011**
r? @petrochenkov

### CODE REVIEW

**Comment by Aaron1011** [comment]
@bors r-

I'm going to add a few of the unusual inner attribute cases from https://github.com/rust-lang/rust/pull/82608

**Comment by Aaron1011** [comment]
@petrochenkov: I've added two new tests to `src/test/ui/proc-macro/inner-attrs.rs`. This is now ready to merge

File: src/test/ui/proc-macro/inner-attrs.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+#[print_target_and_args(first)]
+#[print_target_and_args(second)]
+fn foo() {
+    #![print_target_and_args(third)]
+    #![print_target_and_args(fourth)]
+}
+
+fn bar() {
+    (#![print_target_and_args(fifth)] 1, 2);
```
Comment (petrochenkov) [inline src/test/ui/proc-macro/inner-attrs.rs:19]: Could you add cases for other inner attributes in non-block expressions as well?
(Can be found by searching for `parse_inner_attributes`.)
```rust
[#![attr] 1, 2]

match 0 {
    #![attr]
    _ => {}
}

Struct { #![attr] field: 0 }
```

**Comment by Aaron1011** [comment]
@petrochenkov: I've added more tests

**Comment by Aaron1011** [comment]
I had forgotten to add the proper header to the new tests:

```rust
#![no_std] // Don't load unnecessary hygiene information from std
extern crate std;
```

@bors r=petrochenkov

**Pull request merged**